store/parameter_store.go

```
@@ -0,0 +1,306 @@
+		if err != nil {
+			if i < retryCount {
+				continue
+			} else {
+				return resultsPerTry, err
+			}
+		}
+		for _, result := range resp.Parameters {
+			ident, err := getSecretIDFromParamName(*result.Name)
+			if err != nil {
+				return resultsPerTry, err
+			}
+			resultsPerTry = append(resultsPerTry, ident)
+		}
+		if len(resultsPerTry) >= len(results) {
+			results = resultsPerTry
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return results, nil
+}
+
+// ListAll gets all secrets within a environment (env)>
+func (s *ParameterStore) ListAll(env Environment) ([]SecretIdentifier, error) {
+	return s.List(env, "")
+}
+
+// History gets history for a secret, returning all versions from the store.
+func (s *ParameterStore) History(id SecretIdentifier) ([]SecretMeta, error) {
+	paramName := getParamNameFromName(id)
+	getParamHistoryInput := &ssm.GetParameterHistoryInput{
+		Name: aws.String(paramName),
+	}
+	apiClient := s.ssmClients[Region]
+	results := []SecretMeta{}
+	resp, err := apiClient.GetParameterHistory(getParamHistoryInput)
+	if err != nil {
+		if awsErr, ok := errors.Cause(err).(awserr.Error); ok {
+			if awsErr.Code() == ssm.ErrCodeParameterNotFound {
+				return results, &IdentifierNotFoundError{Identifier: id}
+			}
+		}
+		return results, err
+	}
+	for _, history := range resp.Parameters {
+		results = append(results, SecretMeta{
+			Created: *history.LastModifiedDate,
+			Version: int(*history.Version) - 1, // AWS is 1-indexed, hence subtract 1
+		})
+	}
+	return results, nil
+}
+
+// Delete deletes all versions of a secret
+func (s *ParameterStore) Delete(id SecretIdentifier) error {
+	deleteParameterInput := &ssm.DeleteParameterInput{
+		Name: aws.String(getParamNameFromName(id)),
+	}
+	for _, regionClient := range s.ssmClients {
+		_, err := regionClient.DeleteParameter(deleteParameterInput)
+		// If any region fails, add to the return list of errors and continue.
+		if err != nil {
+			return err
+		}
```

This seems to return an error, rather than add to a list and return that.

---

Addressed!